consensus/src/chained_bft/block_storage/block_store.rs

```diff
@@ -293,16 +293,22 @@ impl<T: Payload> BlockStore<T> {
         // state and on restart, a new execution will agree with it.  A new execution will match
         // the QuorumCert's state on the next restart will work if there is a memory
         // corruption, for example.
-        if let Some(compute_result) = self.get_compute_result(qc.certified_block().id()) {
-            assert_eq!(
-                compute_result.executed_state.state_id,
-                qc.certified_block().executed_state_id(),
-                "We have inconsistent executed state with the executed state from the quorum \
-                 certificate for block {}, will kill this validator and rely on state \
-                 synchronization to try to achieve consistent state with the quorum \
-                 certificate.",
-                qc.certified_block().id(),
-            );
+        match self.get_block(qc.certified_block().id()) {
+            Some(executed_block) => {
+                // TODO: Once we fill in the root compute result, we should remove this
+                if self.root() != executed_block {
+                    assert_eq!(
+                        executed_block.compute_result().executed_state.state_id,
+                        qc.certified_block().executed_state_id(),
+                        "We have inconsistent executed state with the executed state from the quorum \
+                     certificate for block {}, will kill this validator and rely on state \
+                     synchronization to try to achieve consistent state with the quorum \
+                     certificate.",
```

Comment:
nit: this comment is long and ugly :). Can we use something like "QC for block {} has a different execution result than the local computation. Restart and try again."